#! /bin/sh -e

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0


# DP: Fix PR target/41993, [sh] ICE in create_pre_exit, at mode-switching.c

--- a/gcc/mode-switching.c	2009-11-24 09:50:59.000000000 +0900
+++ b/gcc/mode-switching.c	2009-11-24 09:51:02.000000000 +0900
@@ -325,7 +325,14 @@
 		    else
 		      break;
 		    if (copy_start >= FIRST_PSEUDO_REGISTER)
-		      break;
+			{
+				if (!optimize)
+				{
+					last_insn = return_copy;
+					continue;
+				}
+				break;
+			}
 		    copy_num
 		      = hard_regno_nregs[copy_start][GET_MODE (copy_reg)];
 
